docs/design/features/ScalableApproximateCounting.md

```diff
@@ -0,0 +1,185 @@
+
+We next implemented this same technique within the runtime and JIT, encoding the logic above as a new helper call.
+
+Trying this out on some Tech Empower benchmarks where we force the benchmark to execute Tier0 instrumented code, we see significant improvements in server metrics (here on mvc/plaintext, windows, citrine, switchover at 8192):
+
+| Metric (Tier0 instr)   | Racing           | Scalable |         |
+| ---------------------- | ---------------- | -------------------- | ------- |
+| First Request (ms)     |              155 |                  155 |  +0.22% |
+| Requests/sec           |          223,665 |              352,592 | +57.64% |
+| Requests               |        3,372,373 |            5,308,812 | +57.42% |
+| Mean latency (ms)      |            10.60 |                 7.83 | -26.08% |
+| Max latency (ms)       |           237.25 |               279.12 | +17.65% |
+| Read throughput (MB/s) |            34.77 |                54.81 | +57.64% |
+| Latency 50th (ms)      |            10.03 |                 6.68 | -33.42% |
+| Latency 75th (ms)      |            14.69 |                 9.70 | -33.98% |
+| Latency 90th (ms)      |            17.65 |                11.76 | -33.38% |
+| Latency 99th (ms)      |            29.59 |                 0.00 |         |
+
+And in normal processing, the extra accuracy from *Scalable* seems to provide some small benefits to the Tier1 code as well:
+
+| Metric (Tiered PGO)    | Racing      | Scalable        |         |
+| ---------------------- | ----------- | --------------- | ------- |
+| First Request (ms)     |          90 |              89 |  -1.12% |
+| Requests/sec           |   3,083,453 |       3,187,157 |  +3.36% |
+| Requests               |  46,558,728 |      48,125,121 |  +3.36% |
+| Mean latency (ms)      |        0.97 |            0.92 |  -4.61% |
+| Max latency (ms)       |       98.20 |          109.77 | +11.77% |
+| Read throughput (MB/s) |      479.32 |          495.44 |  +3.36% |
+| Latency 50th (ms)      |        0.76 |            0.74 |  -2.76% |
+| Latency 75th (ms)      |        1.08 |            1.05 |  -2.77% |
+| Latency 90th (ms)      |        1.30 |            1.28 |  -2.05% |
+| Latency 99th (ms)      |        0.00 |            0.00 |         |
+
+We haven't yet explored the impact across a wider set of benchmarks or tried varying the switch-over point.
+
+## Summary
+
+We have presented a technique to implement approximately correct counters with relatively low overhead and good scaling properties. The counter accuracy can be tuned by trading off some of the scalability benefit. The counting process uses minimal runtime state: one RNG per physical thread, plus one storage location per counter. The value of the counter is readily available without any post-processing.
+
+This sort of counter seems well-suited for use in our Dynamic PGO instrumentation.
+
+It may be that approximate counting will be useful in other application areas where scala
```

Comment:
Sentence got cut off here - presumably you just mean to say where scalability is needed but small errors are acceptable

Reply:
yep, thanks.